src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,144 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double CAPACITY_MULTIPLIER = 1.5;
+
+    private T[] storage;
+
+    private int currentCapacity;
+    private int size;
+
+    @SuppressWarnings("unchecked")
+    public ArrayList() {
+        this.storage = (T[]) new Object[DEFAULT_CAPACITY];
+        this.currentCapacity = DEFAULT_CAPACITY;
+    }
+
     @Override
     public void add(T value) {
+        if (this.size == this.currentCapacity) {
+            this.resizeStorage();
+        }
 
+        this.storage[this.size] = value;
+        this.size += 1;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index, false);
 
+        if (this.size == this.currentCapacity) {
+            this.resizeStorage();
+        }
+
+        int idx = this.currentCapacity - 1;
+
+        while (idx > index) {
+            this.storage[idx] = this.storage[idx - 1];
+            idx -= 1;
+        }
```

Comment:
Use [System.arraycopy()](https://docs.oracle.com/javase/8/docs/api/java/lang/System.html#arraycopy-java.lang.Object-int-java.lang.Object-int-int-) to move your array elements.